### PR TITLE
Add Clipped images

### LINKED ISSUE #886: Add shaped image components
With the new services pages we introduce a new element which is a shaped image that can appear both on the left or on the right:

Bildschirmfoto 2020-02-14 um 12 35 48

We need to add a component (or 2 components) for this that we can pass in the image we want to show and whether we want the left or right variant (or we build 2 components - one for the left and one for the right variant, depending on how much of the CSS would even be possible to share between the components). Invocation of the component should be similar to `ShapeFeature` (e.g. https://github.com/simplabs/simplabs.github.io/blob/9fde3763e97c2c65ca2348214b9937a888ad3d07/src/ui/components/ShapeFeatureTimify/template.hbs#L1) with passing `@srcset` and `@sizes` attributes.

For the implementation we can use a similar mechanism that we're also using for the [`ShapeFeature` component based on `clip-path`](https://github.com/simplabs/simplabs.github.io/blob/website-revamp/src/ui/components/ShapeFeature/stylesheet.css#L33).

## TODO

- [x] implement the component(s)
- [ ] add to the respective pages in the right places with the correct images

### How exactly are these components expected to behave on small viewports? Do they span the full viewport width? Are they adjusted left or right? [✅]

Mobile design will be addressed in https://github.com/simplabs/simplabs.github.io/issues/905.

### Which exact images are we using on which page?

TBD

## Resources

* Abstract: https://share.goabstract.com/a901d0cd-76b4-4738-b7de-01a24415be15?collectionLayerId=93d765b1-ca1a-41c8-8dbd-66b38317418f&mode=design

### PR DESCRIPTION
**Pull request by marcoow**
This adds the clipped images as introduced with the new services pages.

closes #886

### CODE REVIEW

**Review by marcoow — COMMENTED** [review]
* There is a problem with the clipped images overlapping the content that comes before it:

Bildschirmfoto 2020-03-04 um 10 20 08

* Also at least on the digital products page, 2 of the 3 clipped images disappear on small viewports – I don't think that's intentional?
* The clipped images are missing on the product design, product development, team augmentation and tutoring pages

**Comment by locks** [comment]
@marcoow since you originally published the PR I cannot assign you to review.

**Review by ghislaineguerin — APPROVED** [review]

**Comment by ghislaineguerin** [comment]
@marcoow are you updating the header image for product design? I sent you the link to the photos on the header PR

**Comment by marcoow** [comment]
@ghislaineguerin can you make an issue please?

File: src/ui/components/PageServices/template.hbs [hunk 1]
```diff
@@ -107,15 +118,23 @@
...
     </ArrowLink>
   </div>
-  <aside layout:class="side" offset:class="after-21">
-    <CardPlaybook />
-  </aside>
+  <div layout:class="bleed-right" offset:class="after-16">
+    <div fluid-image:class="clipped-right">
+      <img
+        fluid-image:class="clipped-image"
+        srcset="/assets/images/photos/ui-sketching@600.jpg 600w, /assets/images/photos/ui-sketching@1200.jpg 1200w"
```
Comment (marcoow) [inline src/ui/components/PageServices/template.hbs]: This is not the image we have in Abstract – if we use it here, it repeats on the digital products page which is not good.

File: src/ui/components/PageDigitalProducts/template.hbs [hunk 2]
```diff
@@ -104,6 +124,16 @@
       Involve our senior engineers to quickly establish all required functionalities for your app. Engineering is our true passion, and unmistakable in every product we deliver.
     </p>
   </div>
+  <div layout:class="bleed-left" offset:class="after-12">
+    <div fluid-image:class="clipped-left">
+      <img
+        fluid-image:class="clipped-image"
+        srcset="/assets/images/photos/coding@900.jpg 900w, /assets/images/photos/coding@1800.jpg 1800w"
```
Comment (marcoow) [inline src/ui/components/PageDigitalProducts/template.hbs]: This is not the image we have in Abstract. It's used in some other places so shouldn't be repeated here.

File: src/ui/components/PageProductDesign/template.hbs [hunk 3]
```diff
@@ -66,6 +76,16 @@
       simplabs provides product design expertise throughout the entire product creation process—from rough concept sketches to tested, final implementation.
     </p>
   </div>
+  <div layout:class="bleed-left" offset:class="after-16">
+    <div fluid-image:class="clipped-left">
+      <img
+        fluid-image:class="clipped-image"
+        srcset="/assets/images/photos/tutoring@900.jpg 900w, /assets/images/photos/tutoring@1800.jpg 1800w"
```
Comment (marcoow) [inline src/ui/components/PageProductDesign/template.hbs]: This is the wrong image as well – see Abstract.

File: src/ui/components/PageProductDevelopment/template.hbs [hunk 4]
```diff
@@ -58,6 +58,16 @@
       Talk to an engineer
     </ArrowLink>
   </div>
+  <div layout:class="bleed-right" offset:class="after-16">
+    <div fluid-image:class="clipped-right">
+      <img
+        fluid-image:class="clipped-image"
+        srcset="/assets/images/photos/team-meeting@900.jpg 900w, /assets/images/photos/team-meeting@1800.jpg 1800w"
```
Comment (marcoow) [inline src/ui/components/PageProductDevelopment/template.hbs]: This is the wrong image – see Abstract.

File: src/ui/components/PageProductDevelopment/template.hbs [hunk 5]
```diff
@@ -66,6 +76,16 @@
       Our standards for code quality and engineering are second to none. Our digital product development support includes:
     </p>
   </div>
+  <div layout:class="bleed-left" offset:class="after-16">
+    <div fluid-image:class="clipped-left">
+      <img
+        fluid-image:class="clipped-image"
+        srcset="/assets/images/photos/coding@900.jpg 900w, /assets/images/photos/coding@1800.jpg 1800w"
```
Comment (marcoow) [inline src/ui/components/PageProductDevelopment/template.hbs]: This is the wrong image – see Abstract.

File: src/ui/components/PageTeamAugmentation/template.hbs [hunk 6]
```diff
@@ -47,22 +47,15 @@
...
-      What makes Simplabs engineers unique?
-    </h2>
-    <p typography:class="lead">
-      We are strongly engaged in open source with several team member on the Ember.js core team and a number of libraries that we maintain ourselves. Our experts are internationally known for their work on all parts of the software development process, often sharing their knowledge at conferences and events across the globe.
-    </p>
+  <div layout:class="bleed-right" offset:class="after-16">
+    <div fluid-image:class="clipped-right">
+      <img
+        fluid-image:class="clipped-image"
+        srcset="/assets/images/photos/team-meeting@900.jpg 900w, /assets/images/photos/team-meeting@1800.jpg 1800w"
```
Comment (marcoow) [inline src/ui/components/PageTeamAugmentation/template.hbs]: This is the wrong image – see Abstract.

File: src/ui/components/PageTutoring/template.hbs [hunk 7]
```diff
@@ -10,7 +10,7 @@
       @label="Training & Tutoring"
       @headline="Raise confidence and increase productivity"
       @attachment="image"
-      @srcset="/assets/images/photos/tutoring@600.jpg 600w, /assets/images/photos/tutoring@1200.jpg 1200w"
+      @srcset="/assets/images/photos/collaboration@600.jpg 600w, /assets/images/photos/collaboration@1200.jpg 1200w"
```
Comment (marcoow) [inline src/ui/components/PageTutoring/template.hbs]: This image is now wrong – see Abstract.

File: src/ui/components/PageTutoring/template.hbs [hunk 8]
```diff
@@ -41,14 +41,15 @@
...
-    </h3>
-    <p>
-      Leave us a message and we'll get back to you, establishing a plan together.
-    </p>
-    <FormContact />
+  <div layout:class="bleed-right" offset:class="after-16">
+    <div fluid-image:class="clipped-right">
+      <img
+        fluid-image:class="clipped-image"
+        srcset="/assets/images/photos/tutoring@600.jpg 600w, /assets/images/photos/tutoring@1200.jpg 1200w"
```
Comment (marcoow) [inline src/ui/components/PageTutoring/template.hbs]: This is the wrong image – see Abstract.

**Review by marcoow — COMMENTED** [review]
Most of the images are wrong here. Also, should we not align the images with our grid instead of using arbitrary offsets? E.g.:

Bildschirmfoto 2020-03-10 um 10 19 39

Or is this intentional? I'm just worried we start adding arbitrary offsets everywhere which would destroy all of the rhythm?

**Comment by locks** [comment]
@marcoow they were aligned to the wrong column, should be fixed now:

image

image

File: src/ui/styles/blocks/offset.block.css [hunk 9]
```diff
@@ -32,6 +32,11 @@
   margin-top: 12rem;
 }
 
+.vertical-16 {
```
Comment (marcoow) [inline src/ui/styles/blocks/offset.block.css]: This does not seem to be used?

File: src/ui/components/PageTutoring/template.hbs (hunk 7 above)
Comment (marcoow) [reply src/ui/components/PageTutoring/template.hbs]: This is still wrong – the header image should not change at all in this PR.

**Review by marcoow — COMMENTED** [review]
I think we could give the images a bit more height, at least on desktop? What do you think @ghislaineguerin ? Also please check all images have appropriate dimensions and are optimized. I see some of them are still too big

File: src/ui/styles/blocks/offset.block.css (hunk 9 above)
Comment (locks) [reply src/ui/styles/blocks/offset.block.css]: Removed

**Comment by locks** [comment]
We have resized the code sample image to fit better, and compressed it. Merging this as all images should be of adequate sizes.

File: src/ui/components/PageDigitalProducts/template.hbs [hunk 10]
```diff
@@ -32,7 +32,17 @@
...
+    <div fluid-image:class="clipped-left">
+      <img
+        fluid-image:class="clipped-image"
+        srcset="/assets/images/photos/sketching@600.jpg 600w, /assets/images/photos/sketching@1200.jpg 1200w"
+        sizes="(max-width: 887px) 600px, 1200px)"
+        alt=""
+      />
+    </div>
+  </div>
+  <div layout:class="split-trailing" offset:class="after-16">
```
Comment (marcoow) [inline src/ui/components/PageDigitalProducts/template.hbs:45]: I believe this to be wrong – it was `offset:class="after-21"` before and should still be that as we're separating 2 sections here I think?

File: src/ui/components/PageDigitalProducts/template.hbs [hunk 11]
```diff
@@ -93,6 +103,16 @@
       Explore our design process
     </ArrowLink>
   </div>
+  <div layout:class="bleed-right" offset:class="after-16">
```
Comment (marcoow) [inline src/ui/components/PageDigitalProducts/template.hbs:106]: This would have to  be `offset:class="after-21"` as well.

**Review by marcoow — COMMENTED** [review]
I think there are some unrelated changes to offsets included in this. Also my previous comment still stands and seems to have been ignored before merging:

> I think we could give the images a bit more height, at least on desktop? What do you think @ghislaineguerin?

**Pull request merged**